core/DataAccess/Model.php

```diff
@@ -304,11 +304,12 @@ public function insertRecord($tableName, $fields, $record, $name, $value)
     {
         // duplicate idarchives are Ignored, see https://github.com/piwik/piwik/issues/987
         $query = "INSERT IGNORE INTO " . $tableName . " (" . implode(", ", $fields) . ")
-                  VALUES (?,?,?,?,?,?,?,?)";
+                  VALUES (?,?,?,?,?,?,?,?) ON DUPLICATE KEY UPDATE " . end($fields) . " = ?";
```

Comment:
If the order of the fields get changed we would also need to adjust the order of the values, which is defined below. With the current implementation using `end` the name of the field could change without causing problems. Using the name directly wouldn't allow that. 
But I could change that if you prefer...